blog.tsx

```diff
@@ -285,10 +301,27 @@ export async function handler(
     }
   }
 
+  const sharedHtmlOptions: HtmlOptions = {
+    colorScheme: blogState.theme ?? "auto",
+    lang: blogState.lang,
+    scripts: IS_DEV ? [{ src: "/hmr.js" }] : undefined,
+    links: [
+      { href: canonicalUrl, rel: "canonical" },
+    ],
+  };
+
+  if (blogState.favicon) {
+    sharedHtmlOptions.links?.push({
+      href: blogState.favicon,
+      type: "image/x-icon",
+      rel: "icon",
+    });
+  }
+
   if (pathname === "/") {
     return html({
-      colorScheme: blogState.theme ?? "auto",
```

Comment:
please use the destructuring assignment `{ ...sharedHtmlOption, title: blogstate.title }`